### PR TITLE
Add progress output

### LINKED ISSUE #139: Unable to see the output image after running main.cpp
I cloned the repository and build the solution file. Its been 2 hours and the execution window is still open.
after running through chapter 7 anti aliasing, It takes more time to run the code.

Is there anyway to know how much time its taking to render the ppm image.

### PR DESCRIPTION
**Pull request by hollasch**
Resolves #139

### CODE REVIEW

**Comment by trevordblack** [comment]
Wow! I really like this change.

Should we propagate it into the text as well?
This solution will need slight tweaking when we multithread the code base.

I benchmarked it, and it looks indistinguishable from before

Without change:
```
% time ./inOneWeekend > inoneweekend.ppm

Done.
./inOneWeekend > inoneweekend.ppm  294.00s user 0.15s system 99% cpu 4:54.23 total
```

With Change:
```
% time ./inOneWeekend > inoneweekend.ppm
Scanlines remaining: 0   
Done.
./inOneWeekend > inoneweekend.ppm  294.11s user 0.30s system 99% cpu 4:54.72 total
```

**Comment by hollasch** [comment]
Gah! I remembered and then spaced on synchronizing with the text. I'll add that later, so we'll sit on this pull until that's done.

**Comment by hollasch** [comment]
@trevordblack — let me know what you think about the text changes.

File: books/RayTracingInOneWeekend.html [hunk 1]
```diff
@@ -105,9 +105,9 @@
      fully on at the top. Red and green together make yellow so we should expect the upper right
      corner to be yellow.
 
-Because the file is written to the program output, you'll need to capture the program output.
-Typically this is done from the command-line by redirecting output into a file (using the `>`
-redirection operator), like so:
+<div class='together'>
+Because the file is written to the program output, you'll need to redirect it to the image file.
```
Comment (trevordblack) [inline books/RayTracingInOneWeekend.html]: redirect it to an image file

File: books/RayTracingInOneWeekend.html [hunk 2]
```diff
@@ -106,8 +106,23 @@
...
+Typically this is done from the command-line by using the `>` redirection operator, like so:
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    build\Release\inOneWeekend.exe > image.ppm
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is how things would look on Windows. On Mac or Linux, it would look like this:
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    build/Release/inOneWeekend > image.ppm
```
Comment (trevordblack) [inline books/RayTracingInOneWeekend.html]: the cmake output doesn't qualify as release. It just dumps the file name in the build folder
e.g.
```   
build/inOneWeekend > image.ppm
```

**Comment by trevordblack** [comment]
The addition of "Adding a Progress Indicator" adds a chapter 2.1

No other chapters in _InOneWeekend_ have subchapters.

The only chapters that have subchapters are in _TheNextWeek_ and _TheRestOfYourLife_.
These subchapters are for those books' Acknowledgments subchapter.

We either need to remove the 2.1 for Progress Indicator, or start partitioning other chapters.

**Comment by trevordblack** [comment]
Changes made in `Incorporate review feedback` solve 2/3 problems. See third commit above.

**Comment by hollasch** [comment]
Subchapters would definitely help in many cases. It's just another tool we have available for organization. Feel free to introduces subchapter headers as needed.

**Comment by trevordblack** [comment]
Subchapters would be a hugely significant change.

I'm opening another issue to hash out that discussion separately.

As least for now--as far as this branch goes--I approve of all changes made.
And approve the pull request.

**Pull request merged**